Helm/opensearch/templates/lifecycle-secret.yaml

```diff
@@ -0,0 +1,15 @@
+{{- if and .Values.probes.readiness.enabled .Values.probes.readiness.username .Values.probes.readiness.password }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ template "opensearch.uname" . }}-probes
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ .Chart.Name }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+type: Opaque
+data:
+  READINESS_PASSWORD: {{ .Values.probes.readiness.password | b64enc }}
+  READINESS_USERNAME: {{ .Values.probes.readiness.username | b64enc }}
+{{- end }}
```

Comment:
See previous recommendations. I'd advise removing the enable condition here. As this should always exist.